src/update/UpdateManager.cpp

```diff
@@ -135,11 +135,10 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
                     auto downloadResponse = audacity::network_manager::NetworkManager::GetInstance().doGet(downloadRequest);
 
                     downloadResponse->setRequestFinishedCallback([downloadResponse, this](audacity::network_manager::IResponse*) {
-                        mProgressDialog.reset();
+                        wxTheApp->CallAfter([this]{ mProgressDialog.reset(); });
+                        
```

Comment:
Now I see you already found the problem about threads I already mentioned.  So as I said, use interactive rebase and fix-up the original.  Don't leave a record of mistakes and corrections.